sriovnet_switchdev_test.go

```diff
@@ -352,6 +352,70 @@ func TestGetSfRepresentorErrorNotExistingUplink(t *testing.T) {
 	assert.Contains(t, err.Error(), expectedError)
 }
 
+func TestGetPortIndexFromRepresentor(t *testing.T) {
+	vfReps := []*repContext{
+		{
+			Name:         "p0",
+			PhysPortName: "p0",
+			PhysSwitchID: "c2cfc60003a1420c",
+		},
+		{
+			Name:         "pf0hpf",
+			PhysPortName: "pf0",
+			PhysSwitchID: "c2cfc60003a1420c",
+		},
+		{
+			Name:         "pf0vf10",
+			PhysPortName: "pf0vf10",
+			PhysSwitchID: "fc10d80003a1420c",
+		},
+		{
+			Name:         "pf0sf50",
+			PhysPortName: "pf0sf50",
+			PhysSwitchID: "fc10d80003a1420c",
+		},
+		{
+			Name:         "eth3",
+			PhysPortName: "",
+			PhysSwitchID: "c2cfc60003a1420c",
+		},
+		{
+			Name:         "noswitchdev",
+			PhysPortName: "",
+			PhysSwitchID: "",
+		},
+	}
+	teardown := setupRepresentorEnv(t, "", vfReps)
+	defer teardown()
+
+	var portID int
+	var err error
+
+	portID, err = GetPortIndexFromRepresentor("pf0vf10")
```

Comment:
Thats pretty much repeating itself with different arg to call `GetPortIndexFromRepresentor`

could you refactor this to something more like in: `TestGetVfRepresentorPortFlavour` ?
where test logic is kept generic and you just feed it different test parameters.